src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterFactory.cs

```diff
@@ -32,7 +34,38 @@ public abstract class JsonConverterFactory : JsonConverter
         /// <param name="options">The <see cref="JsonSerializerOptions"/> being used.</param>
         /// <returns>
         /// An instance of a <see cref="JsonConverter{T}"/> where T is compatible with <paramref name="typeToConvert"/>.
+        /// If <see langword="null"/> is returned, a <see cref="NotSupportedException"/> will be thrown.
         /// </returns>
-        public abstract JsonConverter CreateConverter(Type typeToConvert, JsonSerializerOptions options);
+        public abstract JsonConverter? CreateConverter(Type typeToConvert, JsonSerializerOptions options);
```

Comment:
Let's take this offline for discussion -- the only reason we allow null now is because previously collections would display a nice error message with the parent type + property name for unsupported collections. With the new converter model, now collections are converters so we needed a way to do this (or change the tests). 

However if we instead if JsonPath in the NotSupportedException (which should be good enough IMO) then we don't have to do this.